src/bin/ion/commands/generate/utils.rs

```diff
@@ -107,17 +80,49 @@ impl Language for JavaLanguage {
         )
     }
 
-    fn target_type_as_sequence(target_type: &Option<String>) -> Option<String> {
-        target_type.as_ref().map(|target_type_name| {
-            match JavaLanguage::wrapper_class(target_type_name) {
-                Some(wrapper_name) => format!("ArrayList<{}>", wrapper_name),
-                None => format!("ArrayList<{}>", target_type_name),
-            }
-        })
+    fn target_type_as_sequence(
+        target_type: FullyQualifiedTypeReference,
+    ) -> FullyQualifiedTypeReference {
+        match JavaLanguage::wrapper_class(&format!("{}", target_type)) {
+            Some(wrapper_name) => FullyQualifiedTypeReference {
+                type_name: vec![
+                    "java".to_string(),
+                    "util".to_string(),
+                    "ArrayList".to_string(),
+                ],
+                parameters: vec![FullyQualifiedTypeReference {
+                    type_name: vec![wrapper_name],
+                    parameters: vec![],
+                }],
+            },
+            None => FullyQualifiedTypeReference {
+                type_name: vec![
+                    "java".to_string(),
+                    "util".to_string(),
+                    "ArrayList".to_string(),
+                ],
+                parameters: vec![target_type],
+            },
+        }
     }
 
-    fn is_built_in_type(name: &str) -> bool {
-        matches!(name, "int" | "String" | "boolean" | "byte[]" | "double")
+    fn is_built_in_type(type_name: String) -> bool {
```

Comment:
If we're using fully qualified references everywhere, what do we need this function for?